internal/datastore/mysql/datastore.go

```diff
@@ -491,6 +498,21 @@ func (mds *Datastore) seedDatabase(ctx context.Context) error {
 		return fmt.Errorf("seedDatabase: failed to get last inserted id: %w", err)
 	}
 
+	uuidSQL, uuidArgs, err := sb.
+		Insert(mds.driver.Metadata()).
+		Options("IGNORE").
+		Columns(metadataIDColumn, metadataUniqueIDColumn).
+		Values(0, uuid.NewString()).
+		ToSql()
+	if err != nil {
+		return fmt.Errorf("seedDatabase: failed to prepare SQL: %w", err)
+	}
+
+	_, err = tx.ExecContext(ctx, uuidSQL, uuidArgs...)
+	if err != nil {
+		return fmt.Errorf("seedDatabase: failed to insert unique ID: %w", err)
+	}
+
```

Comment:
You probably want to move the whole "UUID seeding" block after want to `if lastInsertID != noLastInsertID {`, since it seems like interleaving 2 different concerns.

You may also want to do the same `lastInsertID` check so you can emit a log message indicating the UUID was successfully inserted